cmd/test.go

```diff
@@ -32,11 +33,82 @@ func getTestConfig() (*models.Test, error) {
 	var doc models.Config
 	err = decoder.Decode(&doc)
 	if err != nil {
-		return nil, fmt.Errorf(Emoji, "failed to decode the keploy-config.yaml. error: %v", err.Error())
+		return nil, err
 	}
 	return &doc.Test, nil
 }
 
+func (t *Test) getTestConfig(path *string, proxyPort *uint32, appCmd *string, testsets *[]string, appContainer, networkName *string, Delay *uint64, passThorughPorts *[]uint, apiTimeout *uint64, globalNoise *models.GlobalNoise, noise *models.TestsetNoise, configPath string) {
+	configFilePath := filepath.Join(configPath, "keploy-config.yaml")
+	if isExist := utils.CheckFileExists(configFilePath); !isExist {
+		t.logger.Info("keploy configuration file not found")
+		return
+	}
+	confTest, err := readTestConfig(configFilePath)
+	if err != nil {
+		t.logger.Error("failed to get the test config from config file")
+		return
+	}
+	if len(*path) == 0 {
+		*path = confTest.Path
+	}
+	if *proxyPort == 0 {
+		*proxyPort = confTest.ProxyPort
+	}
+	if *appCmd == "" {
+		*appCmd = confTest.Command
+	}
+	if len(*testsets) == 0 {
+		*testsets = confTest.TestSets
+	}
+	if *appContainer == "" {
+		*appContainer = confTest.ContainerName
+	}
+	if *networkName == "" {
+		*networkName = confTest.NetworkName
+	}
+	if *Delay == 5 {
```

Comment:
How will this work..? you are not setting delay = 5 right..? dont put any condition here. After calling this function if the delay is empty then assign to delay.

Reply:
delay == 5 is the default value when no value is provided for the flag delay through command line !

Reply:
Yes, first you are looking at delay flag but you will get nothing and you are not even setting the default value as 5 and you call getTestConfig then delay will never be 5. Are you setting the default value as 5 ..?

Reply:
if no delay flag is provided, 5 will be set automatically.

Reply:
Oh my bad, I missed it. It is mentioned 
`testCmd.Flags().Uint64P("delay", "d", 5, "User provided time to run its application")`